lib/plugin-webpack5.js

```diff
@@ -0,0 +1,186 @@
+  new BasicEffectRulePlugin('parser'),
+  new BasicEffectRulePlugin('resolve'),
+  new UseEffectRulePlugin()
+])
+
+const progressiveLoaderModule = require('../progressive-loader/module')
+let vueLoaderPath
+try {
+  vueLoaderPath = require.resolve('vue-loader')
+} catch (err) { }
+
+function isVueLoader(use) {
+  return use.ident === 'vue-loader-options' ||
+    use.loader === 'vue-loader' ||
+    (vueLoaderPath && use.loader === vueLoaderPath)
+}
+
+class VuetifyLoaderPlugin {
+  constructor(options) {
+    this.options = options || {}
+  }
+
+  apply(compiler) {
+    // use webpack's RuleSet utility to normalize user rules
+    const { rules } = compiler.options.module;
+    let vueRules;
+    for (const rawRule of rules) {
+      // skip rules with 'enforce'. eg. rule for eslint-loader
+      if (rawRule.enforce) {
+        continue
+      }
+      // skip the `include` check when locating the vue rule
+      const clonedRawRule = Object.assign({}, rawRule)
+      delete clonedRawRule.include
+
+      const ruleSet = ruleSetCompiler.compile([{
+        rules: [clonedRawRule]
+      }])
+      vueRules = ruleSet.exec({
+        resource: 'foo.vue'
+      })
+      if (vueRules.length > 0) {
+        break;
+      }
+    }
+
+    if (!vueRules.length) {
+      throw new Error(
+        `[VuetifyLoaderPlugin Error] No matching rule for vue-loader found.\n` +
+        `Make sure there is at least one root-level rule that uses vue-loader.`
+      )
+    }
+
+    // find the rules that apply to vue files
+    vueRules = rules.filter(rule => rule.use && rule.use.find(isVueLoader))
+    vueRules.forEach(this.updateRule.bind(this))
+    compiler.options.module.rules = rules;
+  }
+
+  updateRule(rule) {
+    if (this.options.progressiveImages) {
+      const vueLoaderOptions = rule.use.find(isVueLoader).options
+      vueLoaderOptions.compilerOptions = vueLoaderOptions.compilerOptions || {}
+      vueLoaderOptions.compilerOptions.modules = vueLoaderOptions.compilerOptions.modules || []
+      vueLoaderOptions.compilerOptions.modules.push(progressiveLoaderModule)
+
+      const imageRuleIndex = this.rules.findIndex(rule => {
+        return rule.resource &&
+          !rule.resourceQuery &&
+          ['.png', '.jpg', '.jpeg', '.gif'].some(ext => rule.resource(ext))
+      })
+      let imageRule = this.rules[imageRuleIndex]
+
+      const options = typeof this.options.progressiveImages === 'boolean'
+        ? undefined
+        : this.options.progressiveImages
+
+      if (!imageRule) {
+        imageRule = {
+          test: /\.(png|jpe?g|gif|svg|eot|ttf|woff|woff2)(\?.*)?$/,
+          oneOf: [
+            {
+              test: /\.(png|jpe?g|gif)$/,
+              resourceQuery: options ? options.resourceQuery : /vuetify-preload/,
+              use: [
+                {
+                  loader: 'vuetify-loader/progressive-loader',
+                  options
+                },
+                {
+                  loader: 'url-loader',
```

Comment:
All of these loders needs to be resolved up front using `require.resolve`

Ref https://yarnpkg.com/advanced/rulebook#packages-should-only-ever-require-what-they-formally-list-in-their-dependencies